packages/shared-vpc/lib/shared-vpc.ts

```diff
@@ -0,0 +1,125 @@
+import { Construct, CfnOutput } from "@aws-cdk/core";
+import { SubnetType, Vpc } from '@aws-cdk/aws-ec2';
+import { ARecord, CnameRecord, PrivateHostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+
+const DEFAULT_ZONE_RECORD_SUFFIX = 'root';
+
+type Zone = {
+  type: string,
+  target: string,
+  record?: string
+}
+
+export interface SharedVpcProps {
+  /**
+   * The name we should use to create the VPC and prefix it's resources with
+   */
+  vpcName: string
+  /**
+   * The optional CIDR address
+   */
+  cidr?: string,
+  /**
+   * The optional domain to use for the hosted-zone
+   */
+  hostedZoneDomain?: string,
+  /**
+   * The optional zone records as a JSON string
+   * Ex: "[{ "type": "A", "target": "10.6.0.12", "record": "subdomain" }]"
+   */
+  hostedZoneRecords?: string,
```

Comment:
Can we just make this a `Zone[]` ? Instead of a json string and parsing later?

Reply:
Yeah nice, since we're not accepting this from the CLI anymore, we don't need to ask for a JSON string :pray: